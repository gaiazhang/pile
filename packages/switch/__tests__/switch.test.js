import React from 'react';
import { shallow } from 'enzyme';
import { expect as cExpect } from 'chai';
import sinon from 'sinon';
import Switch from '../src';


describe('Switch suite', () => {
  it('should render a Switch', () => {
    const wrapper = shallow(<Switch />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should default the checked property to false.', () => {
    const wrapper = shallow(<Switch />);
    cExpect(wrapper.instance().props.checked).to.equal(false);
  });
  it('click Swtich, checked attribute becomes true, by onChange callback', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(<Switch onChange={onChange} />);
    wrapper.find('[role="switch"]').simulate('click');
    cExpect(onChange).to.have.property('callCount', 1);
  });

  it('switch has disabled state, in the disabled state, onChange event invalid clicks', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(<Switch disabled onChange={onChange} />);
    wrapper.find('[role="switch"]').simulate('click');
    cExpect(onChange).to.have.property('callCount', 0);
  });
});
