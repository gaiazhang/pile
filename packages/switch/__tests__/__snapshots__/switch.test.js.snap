// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch suite should render a Switch 1`] = `
<div
  aria-checked={false}
  className="pile-switch--wrap"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="switch"
  style={
    Object {
      "backgroundColor": "#ccc",
      "borderRadius": "30px",
      "height": "30px",
      "width": "50px",
    }
  }
  tabIndex="0"
>
  <input
    name=""
    type="checkbox"
    value={false}
  />
  <div
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": "28px",
        "height": "28px",
        "left": "1px",
        "marginLeft": 0,
        "width": "28px",
      }
    }
  />
</div>
`;
